### PR TITLE
Search view hierarchy for maps using BFS vs DFS

### LINKED ISSUE #91: Use BFS instead of DFS when locating Maps in the view hierarchy
Currently (v1.2.0), `MapView` instances are added to the processing list using a Depth-First Search (DFS). To ensure that the composite bitmap created by merging the non-`MapView` bitmap with `MapView` bitmaps respects the layering of the view hierarchy, we should replace this DFS with a Breadth-First Search (BFS).

### PR DESCRIPTION
**Pull request by stkent**
In particular, this is important to ensure that when we combine map
bitmaps with the base non-map bitmap, we respect the layering of the
view hierarchy.

Closes #91

### CODE REVIEW

File: bugshaker/src/main/java/com/github/stkent/bugshaker/flow/email/screenshot/maps/MapScreenshotProvider.java [hunk 1]
```diff
@@ -112,15 +116,20 @@ public MapScreenshotProvider(
...
-            // Yes, MapView is a ViewGroup, but I never want to see anyone nesting a MapView
-            // inside another MapView...
-            result.add((MapView) view);
-        } else if (view instanceof ViewGroup) {
-            final ViewGroup viewGroup = (ViewGroup) view;
+        final Queue<View> viewsToProcess = new LinkedList<>();
+        viewsToProcess.add(view);
+
+        while (!viewsToProcess.isEmpty()) {
+            final View viewToProcess = viewsToProcess.remove();
```
Comment (btkelly) [inline bugshaker/src/main/java/com/github/stkent/bugshaker/flow/email/screenshot/maps/MapScreenshotProvider.java:125]: Have you tested this? Usually it is illegal to modify a collection while iterating over it and will result in a [ConcurrentModificationException](https://docs.oracle.com/javase/7/docs/api/java/util/ConcurrentModificationException.html) but it's possible I'm wrong. You can get around this by using an [Iterator](https://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html).

Comment (stkent) [inline bugshaker/src/main/java/com/github/stkent/bugshaker/flow/email/screenshot/maps/MapScreenshotProvider.java:125]: Mmm, good question. Will investigate.

**Comment by stkent** [comment]
Note that the test I just added fails if `locateMapViewsInHierarchy` is replaced with the following code that implicitly uses an iterator:

``` java
@NonNull
@VisibleForTesting
protected final List<MapView> locateMapViewsInHierarchy(@NonNull final View view) {
    final List<MapView> result = new ArrayList<>();

    final Queue<View> viewsToProcess = new LinkedList<>();
    viewsToProcess.add(view);

    for (View viewToProcess : viewsToProcess) {
        if (viewToProcess instanceof MapView && viewToProcess.getVisibility() == VISIBLE) {
            result.add((MapView) viewToProcess);
        } else if (viewToProcess instanceof ViewGroup) {
            final ViewGroup viewGroup = (ViewGroup) viewToProcess;

            for (int childIndex = 0; childIndex < viewGroup.getChildCount(); childIndex++) {
                viewsToProcess.add(viewGroup.getChildAt(childIndex));
            }
        }
    }

    return result;
}
```

**Comment by btkelly** [comment]
Hmm changing the rules... I guess I'll let it slide

**Pull request merged**